x-pack/plugins/apm/server/lib/service_map/get_service_map_from_trace_ids.test.ts

```diff
@@ -5,24 +5,198 @@
  */
 
 import { getConnections } from './get_service_map_from_trace_ids';
-import serviceMapFromTraceIdsScriptResponse from './mock_responses/get_service_map_from_trace_ids_script_response.json';
-import { PromiseReturnType } from '../../../typings/common';
-import { fetchServicePathsFromTraceIds } from './fetch_service_paths_from_trace_ids';
+import { Connection, ConnectionNode } from '../../../common/service_map';
+
+function getConnectionsPairs(connections: Connection[]) {
+  return connections
+    .map((conn) => {
+      const source = `${conn.source['service.name']}:${conn.source['service.environment']}`;
+      const destination = `${conn.destination['service.name']}:${conn.destination['service.environment']}`;
+      if (conn.destination['service.name']) {
+        return `${source} -> ${destination}`;
+      }
+    })
+    .filter((_) => _)
+    .sort();
+}
+
+const paths = [
```

Comment:
Might be better to put this in a `describe('with environments defined')`, similar to what you're doing with missing environments.